django/db/models/base.py

```diff
@@ -402,6 +402,7 @@ class ModelState:
 
 
 class Model(metaclass=ModelBase):
+    _state: ModelState
```

Comment:
We don't include type annotations in core yet https://accounts.google.com/ServiceLogin?service=groups2&passive=1209600&osid=1&continue=https://groups.google.com/d/topic/django-developers/C_Phs05kL1Q/discussion&followup=https://groups.google.com/d/topic/django-developers/C_Phs05kL1Q/discussion&authuser=0

It seems to already be part of stubs https://github.com/typeddjango/django-stubs/blob/9f3b95841b2b0468ce28926b6f0bb2362c65402a/django-stubs/db/models/base.pyi#L12-L27

Reply:
Unfortunately it's not the type annotation that's required - the attribute itself must be present on the class, as unittest.mock introspects the class at runtime (when type hints aren't available).